main.cpp

```diff
@@ -0,0 +1,91 @@
+#include "Header.h"
+#include <iostream>
+using std::cin;
+using std::cout;
+using std::endl;
+using std::cerr;
+
+using std::ostream;
+using std::istream;
+
+#include <cmath>
+using std::pow;
+using std::sqrt;
+
+#include <exception>
+using std::exception;
+
+#include <string>
+using std::string;
+
+#include <fstream>
+using std::ifstream;
+using std::ofstream;
+
+
+template <typename T, const unsigned int SIZE> 
+void print_max(const string &filedir){
+	
+	ifstream input(filedir);
+
+
+
+	Point<T, SIZE> p, pmax;
+
+	try {
+		input >> pmax;
+	}
+	catch(exception &e) {
+		cerr << "Error: unable to read first parenthesis (" << e.what() << ")\n";
+		cerr << "Source code line: " << __LINE__ << endl;
+		cerr << "reading from: " << filedir << endl;
+		cerr << "at position: " << input.tellg() << endl;
+	}
+
+	while (1){
+		try{
+			input >> p;
+		}
+		catch (typename Point<T, SIZE>::EmptyStreamError & e){
+			break; 
+		}
+		catch (typename Point<T, SIZE>::InvalidSymbolError & e){
+			cerr << "Error: ignoring invalid element (" << e.what() << ")\n";
+			cerr << "Source code line: " << __LINE__ << endl;
+			cerr << "reading from: " << filedir << endl;
+			cerr << "at position: " << input.tellg() << endl;
+			while (!input.eof());
```

Comment:
does this work? i don't think the `eof()` method discards any input -- in which case, this would just loop forever (unless `input >> p;` above really had brought us to the end of the file)